bootstrap/internal/cloudinit/cloudinit.go

```diff
@@ -121,3 +124,37 @@ func generate(kind string, tpl string, data interface{}) ([]byte, error) {
 
 	return out.Bytes(), nil
 }
+
+func cleanupAdditionalCloudInit(cloudInitData string) (string, error) {
+	m := make(map[string]interface{})
+
+	err := yaml.Unmarshal([]byte(cloudInitData), m)
+	if err != nil {
+		return "", err
+	}
+
+	// Remove the runcmd section
+	delete(m, "runcmd")
+
+	// Remove the write_files section
+	delete(m, "write_files")
```

Comment:
Also, do we want to create a warning that the extra bootstrap data contained a field that is going to be ignored?

Reply:
Or check in the validating webhook and reject the change?

Reply:
a warning might be useful, however, I don't think we should block the incoming bootstrap object just because there are field that will be ignored, it would be a hassle for the users IMHO.

Reply:
I documented in the API description that some fields would be ignored. Adding a warning involves some changes to the code that do not really make sense because the logger is not available in this nested func. What do you think ?